src/thunder/renderer-backend.cpp

```diff
@@ -140,11 +140,25 @@ struct wpe_renderer_backend_egl_interface thunder_renderer_backend_egl_interface
     // destroy
     [](void* data)
     {
+        if (data != nullptr) {
+            Thunder::Compositor::IDisplay* display = static_cast<Thunder::Compositor::IDisplay*>(data);
+            display->Release();
+        }
     },
     // get_native_display
     [](void* data) -> EGLNativeDisplayType
     {
-        return Thunder::Compositor::IDisplay::Instance(Thunder::DisplayName())->Native();
```

Comment:
I think that this was your leak, that you where trying to solve above, cause the method Thunder::Compositor::IDisplay::Instance(Thunder::DisplayName()) returns a reference counted interface to the callee and the callee is now responsible to Release. By now using the data, the release is no needed, see discussion above..